common/httpx/httpx.go

```diff
@@ -113,6 +113,12 @@ func New(options *Options) (*HTTPX, error) {
 		},
 		DisableKeepAlives: true,
 	}
+	// go clients have issues with java based proxies in case of Tls 1.3 and when the 'Connection: close'
+	// header is present, since the latter is added automatically by the go standard library we opt to lower
+	// max tls version to 1.2 when a proxy is used and most likely matches burp suite
+	if stringsutil.HasSuffixAny(httpx.Options.HTTPProxy, "127.0.0.1:8080", "localhost:8080") {
+		transport.TLSClientConfig.MaxVersion = tls.VersionTLS12
```

Comment:
shouldn't it be better to enable this compatability related behaviour with a flag?

Reply:
I'm just not sure how intuitive it would be to use it. What do you think about performing an automatic connectivity test with the proxy? Then, in case of a `tls: user canceled` error, switching to TLS 1.2?

Reply:
yeah that sounds better

Reply:
found a better solution just by ignoring tls contextual errors